src/model/transaction/splitBlockInContentState.js

```diff
@@ -95,10 +96,22 @@ const splitBlockInContentState = (
   invariant(selectionState.isCollapsed(), 'Selection range must be collapsed.');
 
   const key = selectionState.getAnchorKey();
-  const offset = selectionState.getAnchorOffset();
   const blockMap = contentState.getBlockMap();
   const blockToSplit = blockMap.get(key);
   const text = blockToSplit.getText();
+
+  if (!text) {
+    const blockType = blockToSplit.getType();
+    if (blockType && blockType.endsWith('-list-item')) {
```

Comment:
I think we should be specific about the two list item types:
`if (blockType === 'unordered-list-item' || blockType === 'ordered-list-item')`

Reply:
Agreed! All set.